packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -146,7 +144,6 @@ class TestCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts {
       )
       ..addOption('concurrency',
         abbr: 'j',
-        defaultsTo: math.max<int>(1, globals.platform.numberOfProcessors - 2).toString(),
```

Comment:
Although we are not setting the default value for concurrency in Flutter, we are eventually calling `dart test` which does have a default set for concurrency. As you can see here https://pub.dev/packages/test#test-concurrency, the default is going to be half of the host's CPU cores. ie. for me I have 10, so flutter would have defaulted to 8 while dart defaulted to 5.

So instead of removing the default help text, we should pass the default from dart.

This will prevent people from thinking that they are running on one thread when in reality it is likely running on a few.

Reply:
Actually, I'm not 100% sure what I said above is correct, @christopherfujino, do you know if we are passing the args from the flutter command to `dart test` and using the dart defaults for test? Or do we have our own implementation that doesn't use the defaults?

Reply:
Hmm, then how can we read data from dart_test.yaml? If we are passing a value like `dart test --concurrency sth`, dart test will ignore it, and thus the bug still occurs

Reply:
Perfect, thanks for the pointer @christopherfujino, this is the snippet from the last link above

```dart
  /// How many tests to run concurrently.
  int get concurrency =>
      pauseAfterLoad ? 1 : (_concurrency ?? defaultConcurrency);
```

And the ultimate fall back value is going to be stored in `defaultConcurrency` which is calculated in this file https://github.com/dart-lang/test/blob/master/pkgs/test_core/lib/src/runner/configuration/values.dart#L10-L14

```dart
/// The default number of test suites to run at once.
///
/// This defaults to half the available processors, since presumably some of
/// them will be used for the OS and other processes.
final defaultConcurrency = math.max(1, Platform.numberOfProcessors ~/ 2);
```

So @fzyzcjy, instead of removing the `defaultsTo` parameter, we should probably reference that value using flutter's globals to get the number of processors. Without this, developers may assume they are running on one worker when that may not be the case

Reply:
> So @fzyzcjy, instead of removing the defaultsTo parameter, we should probably reference that value using flutter's globals to get the number of processors. Without this, developers may assume they are running on one worker when that may not be the case

Not sure what you mean. What does "reference that value using flutter's globals" mean? Are we talking about the help text or `defaultsTo` value?

Reply:
I was referring to the defaultsTo value that gets printed along with the help text when you run `flutter test --help`. I'm suggesting that we keep the changes made in this PR but for the defaultsTo value, we should refer to the default value that dart uses, which is half of the number of processors on the developer's machine

Reply:
I only skimmed this change, maybe I'm misunderstanding, but I thought the fix here was to remove `defaultsTo` and let package:test figure it out on its own, so that it will respect `dart_test.yaml`.

Reply:
Yes, that is correct, and @fzyzcjy PR fixes that issue.

My only concern is that flutter users may not know that under the hood `package:test` also has a default value for concurrency that will be applied if the flutter developer does not supply a value for concurrency.

For example, if the user runs `flutter test` in their app without specifying concurrent workers, they may be thinking that it will only run on one worker, when in reality `package:test` is doing something else.

This may not be an issue but I thought giving that information in the help text would be helpful

Reply:
But if we provide a `defaultsTo` value, then how would package:test ever look up dart_test.yaml? It will see that it's always being passed an explicit argument.

Reply:
Ah this is my mistake, I forgot that `defaultsTo` will actually pass a value. Sorry for the confusion!

This should be okay then because we will end up with the same bug. I was really just trying to document that there will likely be more than one worker running for `flutter test`